core/src/main/java/de/danoeh/antennapod/core/sync/SyncService.java

```diff
@@ -69,29 +81,38 @@
     private static final String WORK_ID_SYNC = "SyncServiceWorkId";
     private static final ReentrantLock lock = new ReentrantLock();
 
-    private ISyncService syncServiceImpl;
+    private final Map<String, ISyncService> syncServices = new HashMap<>();
 
     public SyncService(@NonNull Context context, @NonNull WorkerParameters params) {
         super(context, params);
+        syncServices.put(SYNC_PROVIDER_CHOICE_GPODDER_NET, new GpodnetService(AntennapodHttpClient.getHttpClient(),
+                GpodnetPreferences.getHosturl(), GpodnetPreferences.getDeviceID(),
+                GpodnetPreferences.getUsername(), GpodnetPreferences.getPassword()));
+
+        syncServices.put(
+                SYNC_PROVIDER_CHOICE_NEXTCLOUD,
+                new NextcloudSyncService(getApplicationContext(), ClientConfig.USER_AGENT)
+        );
```

Comment:
This means you create an actual instance of all sync providers. I think this is not necessary. How about a method with `switch` statement that returns an instance of the currently selected provider?